gateway/modules/auth/auth.go

```diff
@@ -201,14 +222,24 @@ func (m *Module) IsTokenInternal(token string) error {
 }
 
 func (m *Module) parseToken(token string) (map[string]interface{}, error) {
+	alg := jwt.SigningMethodHS256.Alg()
 	for _, secret := range m.secrets {
 		// Parse the JWT token
 		tokenObj, err := jwt.Parse(token, func(token *jwt.Token) (interface{}, error) {
 			// Don't forget to validate the alg is what you expect:
-			if token.Method.Alg() != jwt.SigningMethodHS256.Alg() {
+			if secret.Alg == jwt.SigningMethodRS256.Alg() {
+				alg = jwt.SigningMethodRS256.Alg()
+			}
+			if token.Method.Alg() != alg {
 				return nil, ErrInvalidSigningMethod
 			}
-
+			if alg == jwt.SigningMethodRS256.Alg() {
+				verifyKey, err := jwt.ParseRSAPublicKeyFromPEM([]byte(secret.PublicKey))
+				if err != nil {
+					return nil, err
+				}
+				return verifyKey, nil
+			}
 			return []byte(secret.Secret), nil
```

Comment:
Simply switch case and return appropriate secret value. Throw error in default case.